llama_index/agent/openai_agent.py

```diff
@@ -40,8 +40,35 @@ def get_function_by_name(tools: List[BaseTool], name: str) -> BaseTool:
     return name_to_tool[name]
 
 
+def call_tool_with_error_handling(
+    tool: BaseTool,
+    input_dict: Dict,
+    error_message: Optional[str] = None,
+    raise_error: bool = False,
+) -> ToolOutput:
+    """Call tool with error handling.
+
+    Input is a dictionary with args and kwargs
+
+    """
+    try:
+        return tool(**input_dict)
+    except Exception as e:
+        if raise_error:
+            raise e
+        error_message = error_message or f"Error: {e!s}"
```

Comment:
nit what is `s` after e!

Reply:
<img width="897" alt="Screenshot 2023-11-18 at 9 23 42 PM" src="https://github.com/run-llama/llama_index/assets/4858925/e7113860-f941-4620-adb7-dfba1fa5df37">